src/errors.rs

```diff
@@ -268,222 +275,344 @@ impl Reportable for DocumentError {
+                        color::Fg(color::Reset)
+                    ),
+                }
+            })
+            .collect();
+        let doc_matches: Vec<String> = match_pairs
+            .iter()
+            .map(|pair| {
+                let content = pair.1
+                    .as_ref()
+                    .unwrap_or(&String::from("<No Match>"))
+                    .to_owned();
+                match ContentMatchPair::is_match(pair) {
+                    true => format!(
+                        "{}{}{}",
+                        color::Fg(color::Green),
+                        content,
+                        color::Fg(color::Reset)
+                    ),
+                    false => format!(
+                        "{}{}{}",
+                        color::Fg(color::Red),
+                        content,
+                        color::Fg(color::Reset)
+                    ),
+                }
+            })
+            .collect();
+        let doc_snippet = doc_node
+            .capabilities
+            .render
+            .as_ref()
+            .ok_or(HowserError::CapabilityError)?
+            .render_commonmark();
+        let rx_snippet = rx_node
+            .capabilities
+            .render
+            .as_ref()
+            .ok_or(HowserError::CapabilityError)?
+            .render_commonmark();
 
-        Ok(LinkError {
+        Ok(TextualContentError {
+            node_type,
             rx_file,
             rx_line,
-            document_file,
-            document_line,
+            doc_file,
+            doc_line,
+            rx_prompts,
+            doc_matches,
+            rx_snippet,
+            doc_snippet,
         })
     }
 }
 
-impl Reportable for LinkError {
+impl Reportable for TextualContentError {
     fn short_msg(&self) -> String {
-        let error_type =
-            ShellText::ErrorColor(Box::new(ShellText::Literal("Link Error".to_string())));
-        let rx_info = ShellText::Underlined(Box::new(ShellText::Literal(format!(
-            "{} line {}",
-            self.rx_file, self.rx_line
-        ))));
-        let document_info = ShellText::Underlined(Box::new(ShellText::Literal(format!(
-            "{} line {}",
-            self.document_file, self.document_line
-        ))));
-        format!(
-            "{} :: {}, {}",
-            error_type.to_string(),
-            rx_info.to_string(),
-            document_info.to_string()
-        )
+        let error_type = format!(
+            "{}Textual Content Error{}",
+            color::Fg(color::Red),
+            color::Fg(color::Reset)
+        );
+        let rx_info = format!(
+            "{}{} line {}{}",
+            style::Underline,
+            self.rx_file,
+            self.rx_line,
+            style::Reset
+        );
+        let doc_info = format!(
+            "{}{} line {}{}",
+            style::Underline,
+            self.doc_file,
+            self.doc_line,
+            style::Reset
+        );
```

Comment:
I feel like I just read this code; is there any way you can take these two error structs and factor out the common bits? The code to deal with them seems more alike than different.